toolz/functoolz.py

```diff
@@ -200,6 +200,7 @@ def __init__(self, *args, **kwargs):
 
         self.__doc__ = getattr(func, '__doc__', None)
         self.__name__ = getattr(func, '__name__', '<curry>')
+        self.__module__ = getattr(func, '__module__', None)
```

Comment:
I've been bitten by those functions before, but I suppose we were bitten by not using them since `__module__` wasn't being set but should have.

Mostly, I want control over how `curry` behaves, and being explicit makes it easier for us to implement and test `cytoolz.curry`.